packages/realm/binding/android/src/main/java/io/realm/react/RealmReactModule.java

```diff
@@ -16,110 +16,103 @@
 
 import com.facebook.react.bridge.JavaScriptContextHolder;
 import com.facebook.react.bridge.ReactApplicationContext;
 import com.facebook.react.bridge.ReactContextBaseJavaModule;
+import com.facebook.react.bridge.ReactMethod;
+import com.facebook.react.internal.featureflags.ReactNativeFeatureFlags;
 import com.facebook.react.module.annotations.ReactModule;
 import com.facebook.react.turbomodule.core.CallInvokerHolderImpl;
 import com.facebook.soloader.SoLoader;
 
 import java.io.IOException;
-import java.net.InetAddress;
-import java.net.NetworkInterface;
-import java.net.SocketException;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.Enumeration;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-import java.util.Locale;
+import java.util.Objects;
 
 @ReactModule(name = RealmReactModule.NAME)
 class RealmReactModule extends ReactContextBaseJavaModule {
     public static final String NAME = "Realm";
-    private static boolean sentAnalytics = false;
 
-    // used to create a native AssetManager in C++ in order to load file from APK
-    // Note: We keep a VM reference to the assetManager to prevent its being
-    //       garbage collected while the native object is in use.
-    //http://developer.android.com/ndk/reference/group___asset.html#gadfd6537af41577735bcaee52120127f4
+    public RealmReactModule(ReactApplicationContext reactContext) {
+        super(reactContext);
+    }
+
+    @Override
+    @NonNull
+    public String getName() {
+        return NAME;
+    }
+
+    /**
+     * Used to create a native AssetManager in C++ in order to load file from APK
+     * Note: We keep a VM reference to the assetManager to prevent its being garbage collected while the native object is in use.
+     * See: http://developer.android.com/ndk/reference/group___asset.html#gadfd6537af41577735bcaee52120127f4
+     */
     @SuppressWarnings("FieldCanBeLocal")
-    private final AssetManager assetManager;
+    private AssetManager assetManager;
 
-    static {
-        try {
-            SoLoader.loadLibrary("realm");
-        } catch (UnsatisfiedLinkError err) {
-            if (err.getMessage().contains("library \"libjsi.so\" not found")) {
-                throw new LinkageError("This version of Realm JS needs at least React Native version 0.66.0", err);
-            }
-            throw err;
-        }
+    @Override
+    public void invalidate() {
+        invalidateCaches();
     }
 
-    public RealmReactModule(ReactApplicationContext reactContext) {
-        super(reactContext);
+    @ReactMethod(isBlockingSynchronousMethod = true)
+    public boolean injectModuleIntoJSGlobal() {
+        // Calling loadLibrary multiple times will be ignored
+        // We do it here instead of statically to allow the app load faster if Realm isn't accessed.
+        // Effectively emulating the behaviour of TurboModules.
+        SoLoader.loadLibrary("realm");
```

Comment:
Should we wrap it in a `try`/`catch` to provide a better error mesage?

Reply:
I imagine the SoLoader will provide sufficient information - we can add to this if we see common patterns emerge, but right now I don't know what I would write to make it more actionable for developers 🤔 It used to be the case that people forgot to `react-native link` but that's a thing of the past.